### PR TITLE
Allow for node upgrade

### LINKED ISSUE #5426: Upgrade `0.9` Cluster to `0.10` Cluster
This issue will cover the necessary steps required to make the server upgradable.  This is currently WIP while I collect more data.

## `0.9x` directory structure
```
├── data
│   └── _internal
│       └── monitor
│           └── 1
├── hh
├── meta
│   ├── id
│   ├── peers.json
│   ├── raft.db
│   └── snapshots
└── wal
    └── _internal
            └── monitor
                └── 1
	                └── _00001.wal

```

`id` content:
```
1
```

`peers.json` content:
```
["localhost:8088"]
```

`SHOW SERVERS`

```
> show servers
id      cluster_addr    raft    raft-leader
1       localhost:8088  true    true
```

## `0.10` directory structure

```
├── data
│   └── _internal
│       └── monitor
│           └── 1
├── hh
├── meta
│   ├── node.json
│   ├── peers.json
│   ├── raft.db
│   └── snapshots
└── wal
    └── _internal
            └── monitor
                └── 1
	                └── _00001.wal
```

`node.json` content:
```
{"ID":2,"MetaServers":["Corys-MacBook-Pro.local:8091"]}
```

`peers.json` content:
```
["Corys-MacBook-Pro.local:8088"]
```

`SHOW SERVERS`

```
> show servers
name: data_nodes
----------------
id      http_addr                       tcp_addr
2       Corys-MacBook-Pro.local:8086    Corys-MacBook-Pro.local:8088


name: meta_nodes
----------------
id      http_addr                       tcp_addr
1       Corys-MacBook-Pro.local:8091    Corys-MacBook-Pro.local:8088
```

## Starting an `0.10` instance after an `0.9` instance:

```
├── data
│   ├── _internal
│   │   └── monitor
│   │       └── 1
├── hh
├── meta
│   ├── id
│   ├── node.json
│   ├── peers.json
│   ├── raft.db
│   └── snapshots
└── wal
    ├── _internal
        └── monitor
            └── 1
                ├── _00001.wal
                └── _00002.wal
```

`id` content:
```
1
```

`node.json` content:
```
{"ID":3,"MetaServers":["Corys-MacBook-Pro.local:8091"]}
```

`peers.json` content:
```
["Corys-MacBook-Pro.local:8088"]
```

`SHOW SERVERS`

```
> show servers
name: data_nodes
----------------
id      http_addr                       tcp_addr
2       localhost:8088                  localhost:8088
3       Corys-MacBook-Pro.local:8086    Corys-MacBook-Pro.local:8088


name: meta_nodes
----------------
id      http_addr                       tcp_addr
1       Corys-MacBook-Pro.local:8091    Corys-MacBook-Pro.local:8088
```

### PR DESCRIPTION
**Pull request by corylanou**
fixes #5426 only for single node.

//cc @jwilder @dgnorton

### CODE REVIEW

File: services/meta/store_fsm.go [hunk 1]
```diff
@@ -146,8 +146,13 @@ func (fsm *storeFSM) applyCreateNodeCommand(cmd *internal.Command, peers []strin
 		}
 	}
 
-	if err := other.CreateDataNode(v.GetHost(), v.GetHost()); err != nil {
-		return err
+	// Get the only meta node
+	if len(other.MetaNodes) == 1 {
```
Comment (jwilder) [inline services/meta/store_fsm.go]: Will this break starting up a brand new server?  How does the `DataNode` get created in that case?

Comment (corylanou) [inline services/meta/store_fsm.go]: fixed

**Comment by jwilder** [comment]
@corylanou Tested it locally and it worked.  Only thing I noticed was that on the upgrade, I see:

```
> show servers
name: data_nodes
----------------
id	http_addr	tcp_addr
1	localhost:8088	localhost:8088


name: meta_nodes
----------------
id	http_addr	tcp_addr
1	localhost:8091	localhost:8088
```

And starting up fresh, I see:
```
> show servers
name: data_nodes
----------------
id	http_addr	tcp_addr
2	localhost:8086	localhost:8088


name: meta_nodes
----------------
id	http_addr	tcp_addr
1	localhost:8091	localhost:8088
```

Seems like second case should match the first?

**Comment by jwilder** [comment]
```
--- FAIL: TestServer_BackupAndRestore (13.98s)
panic: node already exists [recovered]
	panic: node already exists

goroutine 9 [running]:
testing.func·006()
	/usr/local/go/src/testing/testing.go:441 +0x181
github.com/influxdb/influxdb/cmd/influxd/run_test.OpenServer(0xc208021500, 0x0, 0x0, 0x7)
	/root/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server_helpers_test.go:59 +0x145
github.com/influxdb/influxdb/cmd/influxd/run_test.TestServer_BackupAndRestore(0xc208032000)
```

**Comment by jwilder** [comment]
:+1:

**Pull request merged**